integrationtests/as-lucene-directory/src/test/java/org/infinispan/test/integration/as/wildfly/InfinispanModuleMemberRegistrationIT.java

```diff
@@ -128,10 +128,52 @@ public void testNewMemberSearch() throws Exception {
    @Test
    @InSequence(value = 3)
    @OperateOnDeployment("dep.active-2")
+   public void testNewMemberLuceneSearch() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("Peter");
+
+      assertFalse("Expected at least one result after the indexing", search.isEmpty());
+      assertEquals("Lucene search hasn't found a member", "Peter O'Tall", search.get(0).getName());
+   }
+
+   @Test
+   @InSequence(value = 4)
+   @OperateOnDeployment("dep.active-2")
    public void testNonExistingMember() throws Exception {
       List<Member> search = memberRegistration.search("TotallyInventedName");
 
       assertNotNull("Search should never return null", search);
       assertTrue("Search results should be empty", search.isEmpty());
    }
+
+   @Test
+   @InSequence(value = 5)
+   @OperateOnDeployment("dep.active-2")
+   public void testLuceneNonExistingMember() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("TotallyInventedName");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 6)
+   @OperateOnDeployment("dep.active-2")
+   public void testPurgeIndex() throws Exception {
+      memberRegistration.purgeMemberIndex();
+      List<Member> search = memberRegistration.search("Peter");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 7)
+   @OperateOnDeployment("dep.active-2")
+   public void testReIndex() throws Exception {
```

Comment:
The fix is simple: delete the _Resources_ helper, that was a bad idea.

The case of test _testPurgeIndex()_ works because there are no results; the code only blows up when it needs to actually connect to the database to load data in the same transaction (a transaction which is invalid at that point).
Not having any result, it doesn't attempt loading.

I guess we could do a more aggressive validation to have it throw an exception even in that case, just for consistency. Tracking this "enhancement" (if so we can call something which throws more exceptions) as https://hibernate.atlassian.net/browse/HSEARCH-1857